### DISCUSSION TITLE
proxy only what is in router property, rest ignore

### DISCUSSION THREAD

**Discussion by DaliborTrampota** [chimurai/http-proxy-middleware#829] [Q&A]
Hi, 

I'm having issues with setting up the proxy.
I host site and now cdn on the same machine I want the cdn to have cdn subdomain so I proxy it like so. It works but when I visit the page at https.//koknut.xyz it does not work and says "Must provide a proper URL as target" so I figured that I have to add target property so I did 'https://koknut.xyz' but now it says this "[HPM] Error occurred while proxying request koknut.xyz/images/KokNutLogo.png to https://144.172.70.52/ [CERT_HAS_EXPIRED]" I dont know if my cert has expired or not but without the proxy, my https works just fine and so does the site. Why it cant just ignore all other paths/hosts and only proxy the router defined ones?
I want to proxy only /api/webho... and cdn.koknut.xyz

```
app.use(createProxyMiddleware({
    target: 'https://koknut.xyz',
    router: {
        'cdn.koknut.xyz': 'https://koknut.xyz:2087',
        '/api/webhooks/twitch/callback': 'https://koknut.xyz:2096/api/webhooks/twitch/callback',//app.post('/api/webhooks/twitch/callback', createProxyMiddleware(`https://koknut.xyz:2096/api/webhooks/twitch/callback`))
        //'/api/webhooks/twitch/callback': 'http://localhost:2096/api/webhooks/twitch/callback'
    }
}))

```

**Comment by chimurai** [comment]
`router` only allows you to make the `target` dynamic.

To decide which request to proxy or not (ignore or allow), you can use context-matching (v2) or pathFilter (v3):

v2: https://github.com/chimurai/http-proxy-middleware/blob/v2.0.6/recipes/context-matching.md
v3: https://github.com/chimurai/http-proxy-middleware/blob/master/recipes/pathFilter.md (currently in beta)

**Reply by DaliborTrampota** [reply]
Thanks, but I nade my own middleware function using http-proxy package. I got headache trying to figure out how to get it to work.

**Reply by chimurai** [reply]
Sorry to hear you didn't manage. Happy to hear if you have suggestions for improvements.
